test/object-store/sync/client_reset.cpp

```diff
@@ -3704,7 +3704,10 @@ TEST_CASE("client reset with embedded object", "[client reset][local][embedded o
                 actual.test(test_mode == ClientResyncMode::Recover ? expected_recovered : initial);
             });
             int64_t initial_value = initial.dict_values[existing_key]->int_value;
-            int64_t addition = random_int();
+            std::mt19937_64 engine(std::random_device{}());
+            std::uniform_int_distribution<int64_t> rng(-10'000'000'000, 10'000'000'000);
+
+            int64_t addition = rng(engine);
```

Comment:
This is avoiding signed int overflow by generating numbers which won't overflow when added together.